extensions/open-remote-ssh/src/serverSetup.ts

```diff
@@ -0,0 +1,606 @@
+import * as crypto from 'crypto';
+import Log from './common/logger';
+import { getVSCodeServerConfig } from './serverConfig';
+import SSHConnection from './ssh/sshConnection';
+
+export interface ServerInstallOptions {
+    id: string;
+    quality: string;
+    commit: string;
+    version: string;
+    release?: string; // vscodium specific
+    extensionIds: string[];
+    envVariables: string[];
+    useSocketPath: boolean;
+    serverApplicationName: string;
+    serverDataFolderName: string;
+    serverDownloadUrlTemplate: string;
+}
+
+export interface ServerInstallResult {
+    exitCode: number;
+    listeningOn: number | string;
+    connectionToken: string;
+    logFile: string;
+    osReleaseId: string;
+    arch: string;
+    platform: string;
+    tmpDir: string;
+    [key: string]: any;
+}
+
+export class ServerInstallError extends Error {
+    constructor(message: string) {
+        super(message);
+    }
+}
+
+const DEFAULT_DOWNLOAD_URL_TEMPLATE = 'https://github.com/VSCodium/vscodium/releases/download/${version}.${release}/vscodium-reh-${os}-${arch}-${version}.${release}.tar.gz';
```

Comment:
Although this probably won't get used  since we specify the download template url in `product.json`, should we set this to the same value there `"https://github.com/posit-dev/positron/releases/download/${version}/positron-reh-${os}-${arch}-${version}.tar.gz"` anyways?
```suggestion
const DEFAULT_DOWNLOAD_URL_TEMPLATE = 'https://github.com/posit-dev/positron/releases/download/${version}/positron-reh-${os}-${arch}-${version}.tar.gz';
```

Reply:
Good idea!